specs/governance/gov-token.md

```diff
@@ -0,0 +1,101 @@
+# Governance Token
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+  - [Checkpoints](#checkpoints)
+  - [Token Minting](#token-minting)
+  - [Token Burning](#token-burning)
+  - [Voting Power](#voting-power)
+    - [Queries](#queries)
+  - [Delegation](#delegation)
+  - [Supply Cap](#supply-cap)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+| Constants    | Value                                        |
+|--------------|----------------------------------------------|
+| Address      | `0x4200000000000000000000000000000000000042` |
+| Token name   | `Optimism`                                   |
+| Token symbol | `OP`                                         |
+
+`GovernanceToken` is an [ERC20](https://eips.ethereum.org/EIPS/eip-20) token contract that inherits from `ERC20Burnable`,
+`ERC20Votes`, and `Ownable`. It allows token holders to delegate their voting power to other addresses, enabling a representative
+voting system.
+
+The specific implementation details of the voting and delegation logic are inherited from the `ERC20Votes` contract. The
+`GovernanceToken` contract focuses on integrating this functionality with `ERC20Burnable` and adding the minting
+capability.
+
+### Checkpoints
+
+`ERC20Votes` uses checkpoints to track voting power history. A checkpoint is a "snapshot" of the voting power of an
+address at a specific block number.
+
+```solidity
+struct Checkpoint {
+    uint32 fromBlock;
+    uint224 votes;
+}
+```
+
+Checkpoints correrspond to a mapping of an address to an append-only list of checkpoints, which are updated when tokens
+are transferred, minted, burned, or delegated. Generically, updating checkpoints happens by means of the `_moveVotingPower`
+function.
+
+```solidity
+mapping(address => Checkpoint[]) private _checkpoints;
+```
+
+An account's checkpoints' `fromBlock` values are strictly increasing, which means checkpoints at the same block number get
+overwritten by the new one, ensuring there's always at most one checkpoint per block number. This is enforced by the `_writeCheckpoint`
+function.
+
+Additionally, the contract MUST provide functions `numCheckpoints` and `checkpoints` to allow retrieving the number of
```

Comment:
which contract does this line refer to? `GovernanceToken` doesn't provide `numCheckpoints` and `checkpoints` because they are already inherited from `ERC20Votes`